src/main/java/org/reactome/orthoinference/PathwaysInferrer.java

```diff
@@ -73,8 +74,15 @@ private static void createInferredPathwayHierarchy(GKInstance sourceEventInst) t
 
 		for (GKInstance sourcePathwayReferralInst : sourcePathwayReferralInstances)
 		{
+			// Disease TopLevelPathway inference is skipped.
+			if (sourcePathwayReferralInst.equals(diseasePathwayInst)) {
+				logger.info("Pathway referral is " + diseasePathwayInst + " -- skipping Pathway inference");
+				return;
+			}
+
 			logger.info("Generating inferred Pathway: " + sourcePathwayReferralInst);
-			if (inferredEventIdenticals.get(sourcePathwayReferralInst) == null)
+			// Pathways that have been inferred already are skipped, as are Pathways that are only children of the Disease TopLevelPathway.
+			if (inferredEventIdenticals.get(sourcePathwayReferralInst) == null && !InstanceUtilities.onlyInDiseasePathway(sourcePathwayReferralInst))
```

Comment:
It took me a bit of time to understand the first part of the condition.  I wasn't really sure what `inferredEventIdenticals` was since this is what the map is used to check but not what it holds.  Could `inferredEventIdenticals` be renamed to `sourceInstanceToInferredInstance` and the first part of the condition to `!sourceInstanceToInferredInstance.containsKey(sourcePathwayReferralInst)`?  The first part of this condition might be better extracted to its own private method for easy understanding.

```
if (alreadyInferred(sourcePathwayReferralInst) || onlyInDiseasePathway(sourcePathwayReferralInst)) {
    logger.info("Inferred pathway instance {} already exists or only has Disease for a top level pathway", sourcePathwayReferralInst.getExtendedDisplayName());
} else {
   inferPathway(sourcePathwayReferralInst);
}
```

Reply:
OK, I used your example